scene/gui/texture_button.cpp

```diff
@@ -215,17 +217,21 @@ void TextureButton::_notification(int p_what) {
 				}
 
 				_position_rect = Rect2(ofs, size);
+
+				size.width *= hflip ? -1.0f : 1.0f;
+				size.height *= vflip ? -1.0f : 1.0f;
+
 				if (_tile) {
-					draw_texture_rect(texdraw, _position_rect, _tile);
+					draw_texture_rect(texdraw, Rect2(ofs, size), _tile);
```

Comment:
Isn't it problematic to have an inconsistency between the cached `_position_rect` and the actual drawing position?

Reply:
It should be ok since `_position_rect` is only used in `TextureButton::has_point`. The only difference is it's not flipped, but the size and position are the same as the actual drawing rectangle.